packages/adaptive-ui-designer-figma/src/cli/main.ts

```diff
@@ -0,0 +1,147 @@
+import "./dom-shim.js";
+import path from 'path';
+import { Command } from 'commander';
+import inquirer from 'inquirer';
+import { Client  } from './figma-rest-client.js';
+import { LibraryConfig } from './library-config.js';
+import { Logger } from './logger.js';
+import { AnatomyConfiguration } from './anatomy.js';
+import { StyleSheet } from './stylesheet.js';
+import { alphabetize } from './string-utils.js';
+
+const program = new Command();
+const logger = Logger.create();
+const programName = 'Highspot Design-to-Code';
+
+program
+  .name(programName)
+  .description('A CLI tool for generating CSS stylesheets from Figma Library Components')
+  .requiredOption('-l, --library <path>', 'Path to the library configuration file.')
+  .action(main);
+
+interface ProgramOptions {
+  library: string;
+}
+
+async function main({ library }: ProgramOptions) {
+  logger.neutral('Validating library config file...');
+  const configPath = path.resolve(process.cwd(), library);
+  const libraryConfig = await LibraryConfig.create(configPath);
+  if (libraryConfig.valid !== true) {
+    logger.fail('Hmmm... there is something wrong with the config file you provided.');
+    process.exit(1);
+  }
+
+  logger.success('Library config is valid!');
+
+  const patRequest = {
+    type: 'password',
+    name: 'pat',
+    message: 'Please provide your Figma personal access token',
+  };
+
+  const response = await inquirer.prompt([patRequest]);
+  console.log(response.pat, response.pat.length)
+  const client = Client.create({
+    pat: response.pat
+  });
+
+  logger.neutral('Requesting Figma Library.');
+  const libraryComponentsResponse = await client.getFileComponentSets(libraryConfig.file);
+
+  if (libraryComponentsResponse.error || libraryComponentsResponse.status !== 200) {
+    logger.fail(`Accessing Figma library failed with status code ${libraryComponentsResponse.status}`);
+    process.exit(1);
+  }
+
+  logger.success('Your library was successfully retrieved!');
+
+  const { component_sets } = libraryComponentsResponse.meta;
+
+  // This is an arbitrary filter for Polar components
```

Comment:
Let's take this out for now since it should only list published components. There will be a few we won't want to export, but since it's manual it shouldn't be a problem. The list might have some junk, but we can figure out a scalable solution later.

Reply:
Yep good catch.